packages/realm/binding/android/build.gradle

```diff
@@ -0,0 +1,97 @@
+import java.nio.file.Paths
+
+buildscript {
+    repositories {
+        google()
+        mavenCentral()
+        mavenLocal()
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:7.3.1'
+    }
+}
+apply plugin: 'com.android.library'
+
+/**
+ * Highly simplified implementation of the Node modules resolution algorithm to locate the React Native directory.
+ */
+static def findReactNative(startDir) {
+  def currentPath = startDir.toPath().normalize()
+  while (currentPath) {
+    def candidateFile = currentPath.resolve("node_modules/react-native").toFile()
+    if (candidateFile.exists()) {
+      return candidateFile.path
+    }
+    currentPath = currentPath.getParent()
+  }
+  throw new GradleException("Unable to find the React Native directory")
+}
+
+/**
+ * Replaces all occurances of the file path separator with '/' if it isn't already '/'.
+ */
+def sanitizeFilePath(String filePath) {
+   return File.separatorChar === '/' as char ? filePath : filePath.replace(File.separatorChar, '/' as char)
+}
+
+// We're resolving this only to pass it to CMake for a workaround which relies on a private React Native API / header.
+def REACT_NATIVE_ROOT_DIR = sanitizeFilePath(findReactNative(rootProject.projectDir));
+
+def reactNativeArchitectures() {
+    def value = project.getProperties().get("reactNativeArchitectures")
+    return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
+}
+
+
+android {
+    namespace 'io.realm.react'
+    compileSdkVersion rootProject.hasProperty('compileSdkVersion') ? rootProject.compileSdkVersion : 28
+    buildToolsVersion rootProject.hasProperty('buildToolsVersion') ? rootProject.buildToolsVersion : '28.0.3'
+    buildFeatures {
+        prefab = true
+    }
+    defaultConfig {
+        minSdkVersion rootProject.hasProperty('minSdkVersion') ? rootProject.minSdkVersion : 16
+        targetSdkVersion rootProject.hasProperty('targetSdkVersion') ? rootProject.targetSdkVersion : 28
+        externalNativeBuild {
+            cmake {
+                arguments "-DANDROID_STL=c++_shared",
```

Comment:
Try adding `-DCMAKE_CXX_VISIBILITY_PRESET=hidden` to avoid exporting Core symbols, this will reduce a little bit the binary size and help integrate with another SDK compiling Core statically

Reply:
I suspected that was handled by this: https://github.com/realm/realm-js/blob/f84f3321764831b096848002c7c2bfbffe30f3e6/packages/realm/binding/android/CMakeLists.txt#L35

Reply:
With [this change](https://github.com/realm/realm-js/pull/6650/commits/ec9be5e91be5c8aa917a426c8dc393ded8f724bd) I managed to bring down the .so's from ~ 100MB each to:

```
 12M    binding/android/src/main/jniLibs/arm64-v8a/librealm.so
6.9M    binding/android/src/main/jniLibs/armeabi-v7a/librealm.so
 12M    binding/android/src/main/jniLibs/x86/librealm.so
 13M    binding/android/src/main/jniLibs/x86_64/librealm.so
 ```

For some reason, building for release (I test this calling `./gradlew realm:assembleRelease` in the integration test app), I was running `realm:configureCMakeRelWithDebInfo` - now it runs `realm:configureCMakeRelease`.

Reply:
These are the numbers we have for Kotlin as well https://mvnrepository.com/artifact/io.realm.kotlin/cinterop-android/1.16.0 👍